### PR TITLE
:sparkles: implement `ListContributors` for Azure DevOps

### PR DESCRIPTION
**Pull request by JamieMagee**
#### What kind of change does this PR introduce?

Feature

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?

`LisContributors` throws an unsupported error on Azure DevOps

#### What is the new behavior (if this is a feature change)?**

`LisContributors` is implemented for Azure DevOps

- [x] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes

Next step of https://github.com/ossf/scorecard/issues/4177

#### Special notes for your reviewer

I am a little unsure what to put for `Companies` or `Organizations`. For now, I've defaulted to the current Azure DevOps organization.

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
implement `ListContributors` for Azure DevOps
```

### CODE REVIEW

**Comment by JamieMagee** [comment]
The `Contributors` check now passes. Albeit with unexpected results, as I'm not sure what to choose for `Companies` or `Organizations`.

<details>
<summary>Details</summary>

```
$ env SCORECARD_EXPERIMENTAL=1 go run . --repo dev.azure.com/jamiemagee/jamiemagee/_git/scorecard --checks Contributors
Starting [Contributors]
Finished [Contributors]

RESULTS
-------
Aggregate score: 3.0 / 10

Check scores:
|--------|--------------|--------------------------------|-------------------------------------------------------------------------|
| SCORE  |     NAME     |             REASON             |                        DOCUMENTATION/REMEDIATION                        |
|--------|--------------|--------------------------------|-------------------------------------------------------------------------|
| 3 / 10 | Contributors | project has 1 contributing     | https://github.com/ossf/scorecard/blob/main/docs/checks.md#contributors |
|        |              | companies or organizations --  |                                                                         |
|        |              | score normalized to 3          |                                                                         |
|--------|--------------|--------------------------------|-------------------------------------------------------------------------|
```

</details>

File: clients/azuredevopsrepo/contributors.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	contributors []clients.User
+}
+
+func (c *contributorsHandler) init(ctx context.Context, repourl *Repo) {
+	c.ctx = ctx
+	c.once = new(sync.Once)
+	c.repourl = repourl
+	c.errSetup = nil
+	c.getCommits = c.gitClient.GetCommits
+}
```
Comment (spencerschrock) [inline clients/azuredevopsrepo/contributors.go:42]: let's make sure to clear or reset `contributors`

**Review by spencerschrock — COMMENTED** [review]
> The Contributors check now passes. Albeit with unexpected results, as I'm not sure what to choose for Companies or Organizations.

It may be difficult to get meaningful data for this check since everything is inherently contained in a single organization?

**Comment by JamieMagee** [comment]
> It may be difficult to get meaningful data for this check since everything is inherently contained in a single organization?

Exactly. It also returns nonsensical results for imported repositories. A better check might be using email domain to determine organization association. What do you think?

**Review by spencerschrock — APPROVED** [review]
> A better check might be using email domain to determine organization association. What do you think?

This was also suggested in #4415 for GitLab, let's talk about this tomorrow at the meeting.

**Pull request merged**